src/qtfirebaseadmob.h

```diff
@@ -504,28 +503,21 @@ class QtFirebaseAdMobRewardedVideoAd : public QtFirebaseAdMobBase, public fireba
     Q_OBJECT
 
 public:
-    enum PresentationState
-    {
-        PresentationStateHidden,
-        PresentationStateCoveringUI
-    };
-    Q_ENUM(PresentationState)
-
     QtFirebaseAdMobRewardedVideoAd(QObject* parent = 0);
     ~QtFirebaseAdMobRewardedVideoAd();
 
     void setVisible(bool visible) override;
 
 signals:
     void closed();
-    void presentationStateChanged(int state);
+    void presentationStateChanged(QtFirebaseAdMobInterstitial::PresentationState state);
     void rewarded(const QString &type, float value);
 
 public slots:
     void show() override;
 
 private slots:
-    void onPresentationStateChanged(int state);
+    void setPresentationState(QtFirebaseAdMobInterstitial::PresentationState state);
```

Comment:
`RewardedVideoAd` shouldn't use `AdMobInterstitial` ENUMS. I'm sure a re-factoring and splitting classes into their own files help with the overview.